scipy/stats/_stats_py.py

```diff
@@ -10574,49 +10579,44 @@ def linregress(x, y, alternative='two-sided'):
     intercept (95%): 0.616950 +/- 0.544475
 
     """
-    TINY = 1.0e-20
-    x = np.asarray(x)
-    y = np.asarray(y)
+    np = array_namespace(x, y)  # needed for vecdot pre NumPy 2.0
 
-    if x.size == 0 or y.size == 0:
-        raise ValueError("Inputs must not be empty.")
-
-    if np.amax(x) == np.amin(x) and len(x) > 1:
-        raise ValueError("Cannot calculate a linear regression "
-                         "if all x values are identical")
```

Comment:
This is not appropriate in a world of n-d and lazy arrays. Slices with all identical elements produce NaNs.